### ISSUE TITLE
Reporting returning all 0's in version 5.2.4

### ISSUE THREAD

**Issue by pv0s** [jembi/openhim-core-js#1062]
**Issue:**

In version 4 daily and weekly reports used to list all channels and have proper values according to channel activity.
In version 5.2.4 daily and weekly reports list all the channels correctly, but all of them have 0 values even if activity took place. 

**Replication process:**

 I cloned openhim core 5.2.4 locally and got it running through npm.
 I modified the openhim docker compose script to not use a remote image of openhim core, but still use mongo and openhim-console.
 I exposed the mongo port in docker compose to be available to my local openhim core.
 I modified reports.js in openhim-core to log the output of the calculateMetrics function to see the channel information used to populate the channel report map.
 I modified reports.js to log the html that is sent in the email.
 I modified reports.json to not subtract a day and use the current day's report data from 0:00 until 23:59:59.
 I modified default.json to make the daily report run every minute.
 I modified config.md and set the report utc offset to "+0200".
 I logged into openhim console, created 

a user, created a channel and a broken route and did 16 GETs to the channel to cause failures.
    
The report triggered and produced a report with all 0's, the same as on SIT and Production.

I found the following:

When using 5.2.4, with only a current date change (no 1 day subtraction) e.g.

```
    reportPeriodStart = (!utcOffset) ? moment().startOf('day').format(dateTimeFormat) :
      moment().startOf('day').utcOffset(utcOffset).format(dateTimeFormat)
    reportPeriodEnd = (!utcOffset) ? moment().endOf('day').format(dateTimeFormat) :
      moment().endOf('day').utcOffset(utcOffset).format(dateTimeFormat)
```

This produced the following in the channel report map in the calculateMetrics function (notice the empty data array):

2019-11-14T08:32:00.039Z [worker1] info: fetching dailyReport for #Test channel pdevos@***.**.** 5dcd0ca77cd85727c8d2935d
item data is  { channel:
   { updatedBy: { id: 5dcd0c027cd85727c8d292df, name: 'Super User' },
     methods:
      [ 'GET',
        'POST',
        'DELETE',
        'PUT',
        'OPTIONS',
        'HEAD',
        'TRACE',
        'CONNECT',
        'PATCH' ],
     type: 'http',
     allow: [],
     whitelist: [],
     authType: 'public',
     matchContentTypes: [],
     properties: [],
     txViewAcl: [ 'admin' ],
     txViewFullAcl: [ 'admin' ],
     txRerunAcl: [],
     status: 'enabled',
     rewriteUrls: false,
     addAutoRewriteRules: true,
     autoRetryEnabled: false,
     autoRetryPeriodMinutes: 60,
     _id: 5dcd0ca77cd85727c8d2935d,
     routes: [ [Object] ],
     requestBody: true,
     responseBody: true,
     rewriteUrlsConfig: [],
     name: 'Test channel',
     urlPattern: '^/test$',
     matchContentRegex: null,
     matchContentXpath: null,
     matchContentValue: null,
     matchContentJson: null,
     pollingSchedule: null,
     tcpHost: null,
     tcpPort: null,
     alerts: [],
     __v: 0 },
  data: [] }

I then changed the date code to what it was in a previous release:

```
if (flag === 'dailyReport') {
    reportPeriodStart = moment().startOf('day').toDate()
    reportPeriodEnd = moment().endOf('day').toDate()
```

which produced the following (notice that the data array has data):

2019-11-14T09:01:00.032Z [worker1] info: fetching dailyReport for #Test channel pdevos@***.**.** 5dcd0ca77cd85727c8d2935d
item data is  { channel:
   { updatedBy: { id: 5dcd0c027cd85727c8d292df, name: 'Super User' },
     methods:
      [ 'GET',
        'POST',
        'DELETE',
        'PUT',
        'OPTIONS',
        'HEAD',
        'TRACE',
        'CONNECT',
        'PATCH' ],
     type: 'http',
     allow: [],
     whitelist: [],
     authType: 'public',
     matchContentTypes: [],
     properties: [],
     txViewAcl: [ 'admin' ],
     txViewFullAcl: [ 'admin' ],
     txRerunAcl: [],
     status: 'enabled',
     rewriteUrls: false,
     addAutoRewriteRules: true,
     autoRetryEnabled: false,
     autoRetryPeriodMinutes: 60,
     _id: 5dcd0ca77cd85727c8d2935d,
     routes: [ [Object] ],
     requestBody: true,
     responseBody: true,
     rewriteUrlsConfig: [],
     name: 'Test channel',
     urlPattern: '^/test$',
     matchContentRegex: null,
     matchContentXpath: null,
     matchContentValue: null,
     matchContentJson: null,
     pollingSchedule: null,
     tcpHost: null,
     tcpPort: null,
     alerts: [],
     __v: 0 },
  data:
   [ { _id: 5dcd0cae7cd85727c8d2937c,
       channelID: 5dcd0ca77cd85727c8d2935d,
       startTime: 2019-11-13T22:00:00.000Z,
       type: 'd',
       __v: 0,
       successful: 0,
       processing: 0,
       completed: 0,
       completedWithErrors: 0,
       maxResponseTime: 23,
       minResponseTime: 4,
       requests: 16,
       responseTime: 120,
       failed: 16 } ] }

I played around with the config setting "+0200" in both or either of the config and default files and the result was always the same.

[reports.zip](https://github.com/jembi/openhim-core-js/files/3868133/reports.zip)

**Comment by BMartinos** [comment]
Hi @pv0s Thanks for sending through the issue you have picked up with version 5 of the OpenHIM. We will add this issue to our backlog to start the investigation soon.

**Comment by MattyJ007** [comment]
Hi @pv0s This issue should be resolved in [v5.2.5](https://github.com/jembi/openhim-core-js/releases/tag/v5.2.5)

**Comment by pv0s** [comment]
Hi @MattyJ007, thank you. We will test it in SIT and UAT first.
